### PR TITLE
Test/adp 2064 migrate web extension e2e tests

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We must run the web-extension end-to-end test using our local-test-network environment to have more control over the test data and also to improve speed.

# Proposed Solution

Refactor the end-to-end test to be able to run on the local test network, but we must also keep the option to run it on the testnet or mainnet. 

# Important Changes Introduced

A refactor was performed to allow configuring any network when running the test.

### CODE REVIEW

File: packages/e2e/test/web-extension/extension/background/config.ts [hunk 1]
```diff
@@ -9,50 +11,65 @@ import {
...
+  STAKE_POOL_PROVIDER_PARAMS: envalid.json({ default: {} }),
   TX_SUBMIT_PROVIDER: envalid.str({ choices: txSubmitProviderOptions }),
+  TX_SUBMIT_PROVIDER_PARAMS: envalid.json({ default: {} }),
   UTXO_PROVIDER: envalid.str({ choices: utxoProviderOptions }),
-  WALLET_PASSWORD: envalid.str()
+  UTXO_PROVIDER_PARAMS: envalid.json({ default: {} })
 });
+
+const logger = {
+  debug: (x: any) => console.log(x),
```
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/extension/background/config.ts:62]: Might be called with multiple args
```suggestion
  debug: (...x: any[]) => console.log(...x),
```
What is the reason for this change though (why not just use `logger = console`)?

File: packages/e2e/test/web-extension/tsconfig.json [hunk 2]
```diff
@@ -25,6 +25,15 @@
     },
     {
       "path": "../../../wallet/src"
+    },
+    {
+      "path": "../../../cardano-services-client/src"
```
Comment (mkazlauskas) [inline packages/e2e/test/web-extension/tsconfig.json:30]: Should add it to `package.json` dependencies too

**Review by mkazlauskas — CHANGES_REQUESTED** [review]
Looking good, just a few minor changes needed.

10de4e50997ccc815960c24717405c2968bff28f has changes in `core` package - please update commit message, e.g. `refactor: add 'logger' arg to providerFactory`

File: packages/e2e/test/web-extension/extension/background/config.ts (hunk 1 above)
Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/extension/background/config.ts:62]: The extension does not log debug messages if I just do` logger = console` maybe it is the way the extension is being compiled?

Comment (mkazlauskas) [reply packages/e2e/test/web-extension/extension/background/config.ts:62]: It does log debug messages, you just need to unfilter it in 'Default levels'

Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/extension/background/config.ts]: the changes to the core packages were split into a different commit here:
https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/a38fe7dfbee2889497aeb42f5b1847a0c02720e4

File: packages/e2e/test/web-extension/tsconfig.json (hunk 2 above)
Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/tsconfig.json:30]: The dependency was added here: https://github.com/input-output-hk/cardano-js-sdk/blob/10b09f7c68878809e3b8898bd08abc104a706c9c/packages/e2e/package.json#L61

Comment (mkazlauskas) [reply packages/e2e/test/web-extension/tsconfig.json:30]: We should probably remove it from web-extension tsconfig.json too (move to e2e package tsconfig)

Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/tsconfig.json]: addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/7df0b700f623e8cddf629e9c20c5ee619b3c6c3d it was already in the e2e package tsconfig :)

Comment (mkazlauskas) [reply packages/e2e/test/web-extension/tsconfig.json]: I think [ogmios reference](https://github.com/input-output-hk/cardano-js-sdk/pull/356/files#diff-cf4164ff5ad6c088c8d5aaaa7244bb83f76ae7106d5abbeb59861d2691748943R32-R33) should also be removed then

Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/tsconfig.json]: addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/608ba40c8e1327a59eaa81bba4d94f32523d08a8 (core was also duplicated)

File: packages/e2e/src/factories.ts [hunk 3]
```diff
@@ -354,53 +350,52 @@ export const getWallet = async (props: GetWalletProps) => {
   const { wallet } = await setupWallet({
     createKeyAgent: await keyManagementFactory.create(
       props.env.KEY_MANAGEMENT_PROVIDER,
-      updateParamsId(props.idx !== undefined ? props.idx : 0, props.env.KEY_MANAGEMENT_PARAMS)
+      updateParamsId(props.idx !== undefined ? props.idx : 0, props.env.KEY_MANAGEMENT_PARAMS),
+      dummyLogger
     ),
-    createWallet: async (keyAgent) =>
+    createWallet: async (keyAgent: any) =>
```
Comment (rhyslbw) [inline packages/e2e/src/factories.ts:356]:
```suggestion
    createWallet: async (keyAgent: KeyManagement.AsyncKeyAgent) =>
```

File: packages/e2e/src/factories.ts [hunk 4]
```diff
@@ -354,53 +350,52 @@ export const getWallet = async (props: GetWalletProps) => {
   const { wallet } = await setupWallet({
```
Comment (rhyslbw) [inline packages/e2e/src/factories.ts:350]: Is there a reason to not instantiate the logger here:

``` ts
  const logger = props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger;
```
Then pass it to the `keyManagementFactory` and provider factories?

File: packages/e2e/test/web-extension/extension/background/config.ts [hunk 5]
```diff
@@ -9,50 +11,59 @@ import {
...
+const chainHistoryProviderOptions = ['blockfrost', 'http'];
 
 const env = envalid.cleanEnv(process.env, {
   ASSET_PROVIDER: envalid.str({ choices: assetProviderOptions }),
+  ASSET_PROVIDER_PARAMS: envalid.json({ default: {} }),
   BLOCKFROST_API_KEY: envalid.str(),
   CHAIN_HISTORY_PROVIDER: envalid.str({ choices: chainHistoryProviderOptions }),
-  KEY_AGENT: envalid.str({ choices: keyAgentOptions }),
-  LOGGER_MIN_SEVERITY: envalid.str({ choices: loggerMethodNames as string[], default: 'info' }),
+  CHAIN_HISTORY_PROVIDER_PARAMS: envalid.json({ default: {} }),
```
Comment (rhyslbw) [inline packages/e2e/test/web-extension/extension/background/config.ts:41]:
```suggestion
  CHAIN_HISTORY_PROVIDER_PARAMS: envalid.json<TypeMe!>({ default: {} }),
```

File: packages/e2e/test/web-extension/extension/background/config.ts [hunk 6]
```diff
@@ -75,35 +86,70 @@ const blockfrostApi = [
...
-  throw new Error(`ASSET_PROVIDER unsupported: ${env.ASSET_PROVIDER}`);
 })();
 
 export const txSubmitProvider = (async () => {
   switch (env.TX_SUBMIT_PROVIDER) {
     case 'blockfrost': {
       return blockfrostTxSubmitProvider(await blockfrostApi!);
     }
+    case 'http': {
+      return txSubmitHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.TX_SUBMIT_PROVIDER_PARAMS.url, logger });
```
Comment (rhyslbw) [inline packages/e2e/test/web-extension/extension/background/config.ts:122]:
```suggestion
      return txSubmitHttpProvider({ adapter: axiosFetchAdapter, baseUrl: env.TX_SUBMIT_PROVIDER_PARAMS.baseUrl, logger });
```
Better to be consistent. This will be easier when typing the params ENV, as you can pick properties.

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/e2e/src/factories.ts (hunk 4 above)
Comment (AngelCastilloB) [reply packages/e2e/src/factories.ts:350]: The key managers can not be injected with a logger yet (I believe), but the factory interface requires it, so for now I am giving it a dummy logger.

Comment (rhyslbw) [reply packages/e2e/src/factories.ts:350]: This is what I mean:
``` ts
Index: packages/e2e/src/factories.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/e2e/src/factories.ts b/packages/e2e/src/factories.ts
--- a/packages/e2e/src/factories.ts	(revision 84ca9940c0a2af72475c1686c006f9240e64ed3a)
+++ b/packages/e2e/src/factories.ts	(date 1659243657335)
@@ -347,11 +347,12 @@
  * @param props Wallet configuration parameters.
  */
 export const getWallet = async (props: GetWalletProps) => {
+  const logger = props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger;
   const { wallet } = await setupWallet({
     createKeyAgent: await keyManagementFactory.create(
       props.env.KEY_MANAGEMENT_PROVIDER,
       updateParamsId(props.idx !== undefined ? props.idx : 0, props.env.KEY_MANAGEMENT_PARAMS),
-      dummyLogger
+      logger
     ),
     createWallet: async (keyAgent: any) =>
       new SingleAddressWallet(
@@ -360,39 +361,39 @@
           assetProvider: await assetProviderFactory.create(
             props.env.ASSET_PROVIDER,
             props.env.ASSET_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           chainHistoryProvider: await chainHistoryProviderFactory.create(
             props.env.CHAIN_HISTORY_PROVIDER,
             props.env.CHAIN_HISTORY_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           keyAgent,
           networkInfoProvider: await networkInfoProviderFactory.create(
             props.env.NETWORK_INFO_PROVIDER,
             props.env.NETWORK_INFO_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           rewardsProvider: await rewardsProviderFactory.create(
             props.env.REWARDS_PROVIDER,
             props.env.REWARDS_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           stakePoolProvider: await stakePoolProviderFactory.create(
             props.env.STAKE_POOL_PROVIDER,
             props.env.STAKE_POOL_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           stores: props.stores,
           txSubmitProvider: await txSubmitProviderFactory.create(
             props.env.TX_SUBMIT_PROVIDER,
             props.env.TX_SUBMIT_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           ),
           utxoProvider: await utxoProviderFactory.create(
             props.env.UTXO_PROVIDER,
             props.env.UTXO_PROVIDER_PARAMS,
-            props.env.LOGGER_MIN_SEVERITY ? getLogger(props.env.LOGGER_MIN_SEVERITY) : dummyLogger
+            logger
           )
         }
       )
```

File: packages/e2e/test/web-extension/extension/background/config.ts (hunk 5 above)
Comment (rhyslbw) [reply packages/e2e/test/web-extension/extension/background/config.ts:41]: I've created a separate ticket ADP-2097 to address this change, as the scope here is just extending the existing pattern.

File: packages/e2e/test/web-extension/extension/background/config.ts (hunk 6 above)
Comment (AngelCastilloB) [reply packages/e2e/test/web-extension/extension/background/config.ts:122]: addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/23af88930eb60cf2d4c45d5b044142e8c421e62b

File: packages/e2e/src/factories.ts (hunk 4 above)
Comment (AngelCastilloB) [reply packages/e2e/src/factories.ts:351]: addressed here: https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/cd838c89d9b72f7032883f2c52dd0ac5cc18d551

File: packages/e2e/src/factories.ts (hunk 3 above)
Comment (AngelCastilloB) [reply packages/e2e/src/factories.ts]: addressed here: https://github.com/input-output-hk/cardano-js-sdk/commit/cd838c89d9b72f7032883f2c52dd0ac5cc18d551#diff-5afb753068ccef2f896b6f665314aab6ddf56d00350b535a5f45a5df66a02600R357

**Review by rhyslbw — APPROVED** [review]
Looks great @AngelCastilloB

Please ensure to remove the scope from `refactor(core): add 'logger' arg to providerFactory` as part of the squash.

**Review by rhyslbw — APPROVED** [review]
